File: sdks/go/cmd/elydora/plugins/grok_official_test.go
package plugins

import (
	"encoding/json"
	"os"
	"os/exec"
	"reflect"
	"testing"
)

func TestGrokOfficialBinaryReadsManagedTriple(t *testing.T) {
	binary := os.Getenv("ELYDORA_GROK_BINARY")
	if binary == "" {
		t.Skip("ELYDORA_GROK_BINARY is unset")
	}
	fixture := prepareGrokFixture(t, grokFixtureOptions{})
	if err := fixture.plugin.Install(fixture.config); err != nil {
		t.Fatalf("install Grok hooks: %v", err)
	}
	command := exec.Command(binary, "inspect", "--json")
	command.Dir = fixture.projectDir
	command.Env = append(
		os.Environ(),
		"HOME="+fixture.homeDir,
		"USERPROFILE="+fixture.homeDir,
		"GROK_HOME="+fixture.grokHome,
	)
	output, err := command.CombinedOutput()
	if err != nil {
		t.Fatalf("inspect official Grok config: %v\n%s", err, output)
	}
	var report struct {
		Hooks []struct {
			Event    string `json:"event"`
			HookType string `json:"hookType"`
			Target   string `json:"target"`
			Matcher  any    `json:"matcher"`
			Source   struct {
				Type string `json:"type"`
			} `json:"source"`
		} `json:"hooks"`
	}
	if err := json.Unmarshal(output, &report); err != nil {
		t.Fatalf("decode official Grok inspection: %v\n%s", err, output)
	}
	if len(report.Hooks) != 3 {
		t.Fatalf("official Grok hook count = %d\n%s", len(report.Hooks), output)
	}
	wantTargets := map[string]string{}
	settings := readGrokTestObject(t, fixture.configPath)
	for _, item := range []struct{ event, script string }{
		{"PreToolUse", grokGuardScript},
		{"PostToolUse", grokAuditScript},
		{"PostToolUseFailure", grokAuditScript},
	} {
		wantTargets[item.event] = grokTestManagedHandler(
			t,
			settings,
			item.event,
			item.script,
		)["command"].(string)
	}
	actual := map[string]string{}
	for _, hook := range report.Hooks {
		if hook.HookType != "command" || hook.Matcher != nil ||
			hook.Source.Type != "user" {
			t.Fatalf("official Grok hook shape = %#v", hook)
		}
		actual[hook.Event] = hook.Target
	}
	if !reflect.DeepEqual(actual, wantTargets) {
		t.Fatalf("official Grok hooks = %#v, want %#v", actual, wantTargets)
	}
}

File: sdks/node/test/grok-official.test.mjs
import assert from 'node:assert/strict';
import test from 'node:test';
import {
  createFixture,
  runProcess,
} from '../test-support/grok-test-helpers.mjs';

const grokBinary = process.env.ELYDORA_GROK_BINARY;

test('official Grok Build 0.2.106 discovers the installed matchless hook contract', {
  skip: grokBinary ? false : 'set ELYDORA_GROK_BINARY to the official Grok Build executable',
}, async () => {
  const fixture = await createFixture();
  try {
    const install = await fixture.install();
    assert.equal(install.code, 0, install.stderr);
    const environment = {
      HOME: fixture.homeDir,
      USERPROFILE: fixture.homeDir,
      GROK_HOME: fixture.grokHome,
    };
    const version = await runProcess(
      grokBinary,
      ['--version'],
      environment,
      fixture.projectDir,
    );
    assert.equal(version.code, 0, version.stderr);
    assert.match(version.stdout, /grok 0\.2\.106\b/);

    const inspect = await runProcess(
      grokBinary,
      ['inspect', '--json'],
      environment,
      fixture.projectDir,
    );
    assert.equal(inspect.code, 0, inspect.stderr);
    const report = JSON.parse(inspect.stdout);
    assert.equal(report.grokVersion, '0.2.106');
    assert.deepEqual(
      report.hooks
        .map(({ event, hookType, matcher }) => ({ event, hookType, matcher }))
        .sort((left, right) => left.event.localeCompare(right.event)),
      [
        { event: 'PostToolUse', hookType: 'command', matcher: null },
        { event: 'PostToolUseFailure', hookType: 'command', matcher: null },
        { event: 'PreToolUse', hookType: 'command', matcher: null },
      ],
    );
  } finally {
    await fixture.close();
  }
});
